satellite/gracefulexit/endpoint.go

```diff
@@ -432,33 +464,81 @@ func (endpoint *Endpoint) processIncomplete(ctx context.Context, stream processS
 		path:             incomplete.Path,
 		pieceSize:        pieceSize,
 		satelliteMessage: transferMsg,
+		rootPieceID:      remote.RootPieceId,
+		pieceNum:         incomplete.PieceNum,
 	})
 
 	return nil
 }
 
-func (endpoint *Endpoint) handleSucceeded(ctx context.Context, pending *pendingMap, nodeID storj.NodeID, message *pb.StorageNodeMessage_Succeeded) (err error) {
+func (endpoint *Endpoint) handleSucceeded(ctx context.Context, pending *pendingMap, exitingNodeID storj.NodeID, message *pb.StorageNodeMessage_Succeeded) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	if message.Succeeded.GetOriginalOrderLimit() == nil {
-		return Error.New("original order limit cannot be nil.")
+
+	originalPieceID := message.Succeeded.OriginalPieceId
+
+	transfer, ok := pending.get(originalPieceID)
+	if !ok {
+		endpoint.log.Error("could not find transfer item in pending queue", zap.Stringer("piece ID", originalPieceID))
+		return Error.New("could not find transfer item in pending queue")
```

Comment:
Should all the errors returned by `handleSucceeded` be wrapped with rpc error?

Reply:
Maybe instead of directly returning rpc error from this method, we can create error class and return that. Then on line 335, we can have a switch statement to return corresponding rpc error based on different error classes